### ISSUE TITLE
[Bug]: No module named 'llama_index.core.llms.generic_utils'

### ISSUE THREAD

**Issue by MudassirAqeelAhmed** [run-llama/llama_index#11071]
### Bug Description

I installed llama-index on google colab notebook.
`!pip install llama-index-embeddings-anyscale`
`!pip install -U llama-index llama-index-core llama-index-llms-openai`

I'm trying to import

`from llama_index.embeddings.anyscale import AnyscaleEmbedding`

I'm getting this error

**ModuleNotFoundError: No module named 'llama_index.core.llms.generic_utils'**

### Version

llama-index==0.10.10

### Steps to Reproduce

Just install llama-index, and llama-index-embeddings-anyscale on colab and import

``from llama_index.embeddings.anyscale import AnyscaleEmbedding``

### Relevant Logs/Tracbacks

```shell
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
[<ipython-input-7-6245d537c86e>](https://localhost:8080/#) in <cell line: 21>()
     19 from langchain_core.documents import Document
     20 from langchain.vectorstores.utils import maximal_marginal_relevance
---> 21 from llama_index.embeddings.anyscale import AnyscaleEmbedding
     22 from itertools import islice
     23 from typing import Iterable, Iterator, List, TypeVar, Dict, Tuple, Union, Optional

2 frames
[/usr/local/lib/python3.10/dist-packages/llama_index/embeddings/anyscale/utils.py](https://localhost:8080/#) in <module>
      1 from typing import Optional, Tuple
      2 
----> 3 from llama_index.core.llms.generic_utils import get_from_param_or_env
      4 
      5 DEFAULT_ANYSCALE_API_BASE = "https://api.endpoints.anyscale.com/v1"

ModuleNotFoundError: No module named 'llama_index.core.llms.generic_utils'

---------------------------------------------------------------------------
NOTE: If your import is failing due to a missing package, you can
manually install dependencies using either !pip or !apt.

To view examples of installing some common dependencies, click the
"Open Examples" button below.
---------------------------------------------------------------------------
```

**Comment by hkrsmk** [comment]
When I tried this, issue has been fixed in llama-index-core but the code hasn't changed in the actual package itself when I checked the filepath. Might require updating of pyproject.toml but it's gonna be a pain. You can try running it as a py script and manually changing for now

**Comment by logan-markewich** [comment]
@MudassirAqeelAhmed also update the openai embeddings -- `pip install -U llama-index-embeddings-openai`

**Comment by MudassirAqeelAhmed** [comment]
![image](https://github.com/run-llama/llama_index/assets/94457015/09366c95-45e3-42ad-8bb4-6efd8b95bcf6)


> @MudassirAqeelAhmed also update the openai embeddings -- `pip install -U llama-index-embeddings-openai`

**Comment by MudassirAqeelAhmed** [comment]
I'm running this on notebook.

**Comment by ravi03071991** [comment]
```python
pip uninstall llama-index  # remove any global
python -m venv venv
source venv/bin/activate
pip install llama-index llama-index llama-index-embeddings-openai
pip install ipykernel
python -m ipykernel install --user --name=my_venv --display-name="my_venv"
```
and then select my_venv kernal in your notebook

**Comment by HarmeetSingh07** [comment]
hi @ravi03071991,
I'm facing the same issue and tried updating llama-index but latest llama-index depends on onnxruntime 1.17.0 and this only runs on python 3.12(I think so). Is there a way around this to run:
from llama_index.llms.anyscale import Anyscale

**Comment by urigott** [comment]
I managed to solve this problem by coping _generic_utils.py_ from _llama_index/core/base/llms_ into _llama_index/core/llms_

**Comment by logan-markewich** [comment]
Or you could have installed in a fresh env /upgraded your deps 😀 👍🏻

**Comment by urigott** [comment]
> Or you could have installed in a fresh env /upgraded your deps 😀 👍🏻

For some reason it didn't work for me 🤷

**Comment by manjunathshiva** [comment]
> I managed to solve this problem by copying _generic_utils.py_ from _llama_index/core/base/llms_ into _llama_index/core/llms_

Thanks you!  This worked for me as well! No other methods mentioned worked!  Thank you!

**Comment by PaulBFB** [comment]
> I managed to solve this problem by copying _generic_utils.py_ from _llama_index/core/base/llms_ into _llama_index/core/llms_

I had the exact same issue, this worked for me as well, thank you!

**Comment by ycd** [comment]
Pretty much every integration I try to use with LLamaIndex ends up being problematic 😭

**Comment by logan-markewich** [comment]
@ycd probably more productive to share the issues and get a resolution :)

**Comment by ycd** [comment]
To be fair I was just yapping, solved it already but to be more specific and add a little bit context; I encountered like 3 integrations/projects that has obsolete documentation that does not 'just works', I even ended up monkey-patching some of them to get it working(not this one).

**Comment by ycd** [comment]
Okay, I was using `llama-index-core=0.10.5`,I added Anthropic provider to my project, it raised also this error. 

```python
    from llama_index.llms.anthropic.base import Anthropic
  File "/Users/yagu/wope/castor/env/lib/python3.11/site-packages/llama_index/llms/anthropic/base.py", line 21, in <module>
    from llama_index.core.base.llms.generic_utils import (
ModuleNotFoundError: No module named 'llama_index.core.base.llms.generic_utils'
```

This time, I decided give your suggestion a try, bombed my env, created a new one, and there it is, my env is conflicting now.


```python
The conflict is caused by:
    The user requested llama-index-core==0.10.3
    llama-index 0.10.5 depends on llama-index-core<0.11.0 and >=0.10.0
    llama-index-agent-openai 0.1.1 depends on llama-index-core<0.11.0 and >=0.10.1
    llama-index-callbacks-langfuse 0.1.2 depends on llama-index-core<0.11.0 and >=0.10.8
```

Okay, great, let's upgrade to `0.10.8`: 

```python
    from llama_index.core.prompts.base import (
  File "/Users/yagu/wope/castor/env/lib/python3.11/site-packages/llama_index/core/prompts/base.py", line 37, in <module>
    from llama_index.core.llms.base import BaseLLM
  File "/Users/yagu/wope/castor/env/lib/python3.11/site-packages/llama_index/core/llms/__init__.py", line 12, in <module>
    from llama_index.core.llms.custom import CustomLLM
  File "/Users/yagu/wope/castor/env/lib/python3.11/site-packages/llama_index/core/llms/custom.py", line 19, in <module>
    from llama_index.core.llms.llm import LLM
  File "/Users/yagu/wope/castor/env/lib/python3.11/site-packages/llama_index/core/llms/llm.py", line 43, in <module>
    from llama_index.core.prompts import BasePromptTemplate, PromptTemplate
ImportError: cannot import name 'BasePromptTemplate' from partially initialized module 'llama_index.core.prompts' (most likely due to a circular import)
```

Then, I got circular import error, found your https://github.com/run-llama/llama_index/pull/11032, then decided to upgrade `0.10.15`, which I one by one resolved conflicts like the following:

```python
ERROR: Cannot install -r requirements.txt (line 70) and llama-index-agent-openai==0.1.1 because these package versions have conflicting dependencies.

The conflict is caused by:
    The user requested llama-index-agent-openai==0.1.1
    llama-index 0.10.15 depends on llama-index-agent-openai<0.2.0 and >=0.1.4
```

It's fine I can do that, which I resolved all conflicts, then even in 0.10.15, it still hits me up with this:

```python
    from llama_index.llms.gemini.base import Gemini
  File "/Users/yagu/wope/castor/env/lib/python3.11/site-packages/llama_index/llms/gemini/base.py", line 20, in <module>
    from llama_index.core.utilities.gemini_utils import (
ModuleNotFoundError: No module named 'llama_index.core.utilities.gemini_utils'
```

**Comment by logan-markewich** [comment]
@ycd what are your project reqs? It seems like at this point I would just start a fresh venv with latest versions of things
